### PR TITLE
feat: move route métier + typing + tsoa

### PR DESCRIPTION
**Pull request by alanlr**

### CODE REVIEW

**Review by kevbarns — APPROVED** [review]
👍

File: server/src/http/controllers/metiers/metiers.controller.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import { TGetAppellationsRomesResponseSuccess, TGetMetiersEnrichisResponseSuccess, TGetMetiersResponseSuccess } from "./type.js"
+
+@Tags("Metiers")
+@Route("/api/v1/metiers")
+export class MetiersController extends Controller {
+  /**
+   * Retourne la liste des métiers associé à une formation identifiée par son CFD dans le chemin de l'appel
+   * @returns {Promise<TMetiersResponseSuccess | TResponseError>}
+   */
+  @Response<"List unavailable">(500)
```
Comment (rimiti) [inline server/src/http/controllers/metiers/metiers.controller.ts]: Etrange de ne pas retourner un Internal Server Error.
Mais deal.

File: server/src/services/catalogue.service.ts [hunk 2]
```diff
@@ -312,3 +315,102 @@ export const getFormationsFromCatalogueMe = async ({
...
+      body: {
+        query: {
+          bool: {
+            must: mustTerm,
+          },
+        },
+      },
+    })
+
+    //throw new Error("BOOM");
```
Comment (rimiti) [inline server/src/services/catalogue.service.ts]: to drop

File: server/src/services/catalogue.service.ts [hunk 3]
```diff
@@ -312,3 +315,102 @@ export const getFormationsFromCatalogueMe = async ({
...
+    if (!romes.length) {
+      result.error = "No training found"
+    }
+
+    return result
+  } catch (err) {
+    const error_msg = _.get(err, "meta.body", err.message)
+    console.error("Error getting trainings from romes ", error_msg)
+    if (_.get(err, "meta.meta.connection.status") === "dead") {
+      console.error("Elastic search is down or unreachable")
```
Comment (rimiti) [inline server/src/services/catalogue.service.ts]: it's maybe better to use the logger instead of consoles.

**Review by rimiti — APPROVED** [review]

**Pull request merged**